go/webpack/cbor/cbor_writer.go

```diff
@@ -100,96 +103,140 @@ func encodedSize(i uint64) int {
 	return 8
 }
 
-func (item *compoundItem) encodeInt(t Type, i int) {
-	item.encodeInt64(t, uint64(i))
+func (ci *compoundItem) encodeInt(t Type, i int) {
+	ci.encodeInt64(t, uint64(i))
 }
-func (item *compoundItem) encodeInt64(t Type, i uint64) {
-	item.encodeSizedInt64(encodedSize(i), t, i)
+func (ci *compoundItem) encodeInt64(t Type, i uint64) {
+	ci.encodeSizedInt64(encodedSize(i), t, i)
 }
-func (item *compoundItem) encodeSizedInt64(size int, t Type, i uint64) {
-	item.elements++
+func (ci *compoundItem) encodeSizedInt64(size int, t Type, i uint64) {
+	ci.elements++
 
 	switch size {
 	case 0:
-		item.Write([]byte{byte(t) | byte(i)})
+		ci.Write([]byte{byte(t) | byte(i)})
 	case 1:
-		item.Write([]byte{byte(t) | 24, byte(i)})
+		ci.Write([]byte{byte(t) | 24, byte(i)})
 	case 2:
-		item.Write([]byte{byte(t) | 25, byte(i >> 8), byte(i)})
+		ci.Write([]byte{byte(t) | 25, byte(i >> 8), byte(i)})
 	case 4:
-		item.Write([]byte{byte(t) | 26,
+		ci.Write([]byte{byte(t) | 26,
 			byte(i >> 24), byte(i >> 16), byte(i >> 8), byte(i)})
 	case 8:
-		item.Write([]byte{byte(t) | 27,
+		ci.Write([]byte{byte(t) | 27,
 			byte(i >> 56), byte(i >> 48), byte(i >> 40), byte(i >> 32),
 			byte(i >> 24), byte(i >> 16), byte(i >> 8), byte(i)})
 	default:
 		panic(fmt.Sprintf("Unexpected CBOR item size: %v", size))
 	}
 }
 
-func (item *compoundItem) AppendUint64(i uint64) {
-	item.encodeInt64(TypePosInt, i)
+func (ci *compoundItem) AppendUint64(i uint64) {
+	ci.encodeInt64(TypePosInt, i)
 }
 
 // AppendFixedSizeUint64 always uses the 8-byte encoding for this uint64.
-func (item *compoundItem) AppendFixedSizeUint64(i uint64) {
-	item.encodeSizedInt64(8, TypePosInt, i)
+func (ci *compoundItem) AppendFixedSizeUint64(i uint64) {
+	ci.encodeSizedInt64(8, TypePosInt, i)
 }
-func (item *compoundItem) AppendInt64(i int64) {
+
+func (ci *compoundItem) AppendInt64(i int64) {
 	if i < 0 {
-		item.encodeInt64(TypeNegInt, uint64(-1-i))
+		ci.encodeInt64(TypeNegInt, uint64(-1-i))
 	} else {
-		item.encodeInt64(TypePosInt, uint64(i))
+		ci.encodeInt64(TypePosInt, uint64(i))
+	}
+}
+
+func (ci *compoundItem) AppendBytes(bs []byte) {
+	ci.encodeInt(TypeBytes, len(bs))
+	ci.Write(bs)
+}
+
+type BytesWriter struct {
+	item
+	remainingSize int64
+}
+
+func (bw *BytesWriter) Write(p []byte) (int, error) {
+	n, err := bw.item.Write(p)
+	bw.remainingSize -= int64(n)
```

Comment:
According to https://golang.org/pkg/bufio/#Writer.Write and https://golang.org/pkg/io/#Writer, no: Go uses the err return instead of negative lengths to convey errors.